core/modules/filter/filter.module

```diff
@@ -692,20 +697,26 @@ function filter_modules_disabled($modules) {
  *   (optional) If provided, only those formats that are allowed for this user
  *   account will be returned. All formats will be returned otherwise. Defaults
  *   to NULL.
+ * @param bool $include_disabled
+ *   (optional) Whether or not to include disabled formats in the return array.
+ *   Defaults to FALSE.
+ * @param bool $reset
+ *   (optional) Whether or not to bypass the cache and reload all formats.
+ *   Defaults to FALSE.
  *
  * @return stdClass[]
  *   An array of text format objects, keyed by the format ID and ordered by
  *   weight.
  *
  * @see filter_formats_reset()
  */
-function filter_formats($account = NULL) {
+function filter_formats($account = NULL, $include_disabled = FALSE, $reset = FALSE) {
```

Comment:
I think you can eliminate the `$reset` parameter, as it seems `filter_formats_reset()` provides the same functionality.

Reply:
You mean that we should be specifically calling `filter_formats_reset()` before `filter_formats()` if the intention was to bypass cache?

Reply:
also note if `filter_formats(NULL,, TRUE)` is called, any subsequent calls on the same page load will ignore `$include_disabled` because thats called inside the cache loop.